### DISCUSSION TITLE
Batch transactions

### DISCUSSION THREAD

**Discussion by kordano** [replikativ/datahike#403] [Ideas]
### Describe the feature you would like to request

Batch multiple transactions to Datahike before running I/O to the backend.

### Describe the solution you would like

Add a configurable flag that defines the batch size, then flush the trees after hitting the limit or time limit.

### Describe alternatives you've considered

none

**Comment by kordano** [comment]
@whilo did you have any thoughts on that? I'll try to sketch a prototype for it with just changes in the connector without the db record or the transaction itself knowing anything about it.

**Comment by kordano** [comment]
The first phase could be a simple manual setting where the user could decided whether the next transactions will be flushed. I'll draft a first version for that in a [branch](https://github.com/replikativ/datahike/tree/403-batch-transactions).

**Comment by whilo** [comment]
I agree that this is a reasonable first step. Ultimately I think users don't want to be bother though and we should have adaptive auto-batching until the transactor overflows.
